.github/workflows/main.yml

```diff
@@ -91,24 +91,78 @@ jobs:
     - name: Configure CMake
       if: "matrix.platform.cmake"
       run: |
-        cmake -B build \
+        set -- \
           -DBUILD_SHARED_LIBS=ON \
           -DCMAKE_BUILD_TYPE=Release \
           -DSDL2IMAGE_SAMPLES=ON \
+          -DSDL2IMAGE_TESTS_INSTALL=ON \
           -DSDL2IMAGE_JXL=ON \
           -DSDL2IMAGE_TIF=ON \
           -DSDL2IMAGE_WEBP=ON \
+          ${NULL+}
+
+        if [ "${{ runner.os }}" = Linux ]; then
+            # mingw-w64-*-SDL2 doesn't have SDL_test, and the macOS CMake
+            # build can't find it (see #260), so we only enable build-time
+            # tests for Linux at the moment.
+            set -- "$@" -DSDL2IMAGE_TESTS=ON
+            # If we dlopen() vendored soft dependencies, the test won't
+            # find them: https://github.com/libsdl-org/SDL_image/issues/259
+            set -- "$@" -DSDL2IMAGE_DEPS_SHARED=OFF
+        fi
+
+        cmake -B build \
+          "$@" \
           ${{ matrix.platform.cmake }}
     - name: Build
       if: "matrix.platform.cmake"
       run: cmake --build build/ --config Release --verbose
+    - name: Run build-time tests
+      if: "runner.os == 'Linux' && matrix.platform.cmake"
+      run: |
+        set -eu
+
+        # Expect to support all formats except AVIF in these builds
+        export SDL_IMAGE_TEST_REQUIRE_LOAD_AVIF=0
```

Comment:
For a future pr, I suppose it is ok to move these environment variables to CMake?
[`ENVIRONMENT` property on tests](https://cmake.org/cmake/help/latest/prop_test/ENVIRONMENT.html)

Reply:
No, not really: these are here as a way for the CI to demand that PNG is supported, even if the build system has become confused and disabled PNG due to failed auto-detection or a build system bug.

Specifically, I wanted these for Debian, to detect regressions like the one introduced in 3af38ea and fixed in d25c532 (which accidentally disabled PNG saving in the packages I maintain in Debian experimental). I want to be able to set `SDL_IMAGE_TEST_REQUIRE_LOAD_PNG=1` in the Debian CI, and have that make the test fail if it can't load PNGs.

Reply:
If the CMake build system was responsible for setting these variables, then a bug in the build system that accidentally disabled PNG wouldn't be detected, because the test (which would be told by the build system that PNG was disabled) would be consistent with the implementation (which genuinely did disable PNG).

Reply:
Understood, as a packager it's understandable to not trust the build system to do *the right thing*.
(we should document this)